crates/cli/src/diff.rs

```diff
@@ -0,0 +1,47 @@
+use anyhow::Result;
+use marzano_util::position::{FileRange, Position, RangeWithoutByte, UtilRange};
+use std::{fs::File, io::Read, path::PathBuf, str::FromStr};
+
+pub fn parse_modified_ranges(diff_path: &PathBuf) -> Result<Vec<FileRange>> {
```

Comment:
I think this function should be split in two, parse_modified_ranges should take in a string and the diff reading can be its own function.